src/scope.js

```diff
@@ -110,7 +110,7 @@ var Scope = {
             assert(j !== -1, `unbalanced []: ${str}`)
             name = str.slice(i + 1, j)
             if (!lexical.isInteger(name)) { // foo[bar] vs. foo[1]
-              name = this.get(name)
```

Comment:
What if `this.get(name)` returns `undefined`? can you provide more test cases here

Reply:
If `this.get(name)` returns `undefined`, `name = this.get(name).toString()` will throw an error. 
We can change `name = this.get(name)` to `name = (this.get(name) ? this.get(name) : '').toString()`. 
Thus, if `this.get(name)` returns `undefined`, `name` will be assigned `''`, and no error would be thrown.